source/StarField.cpp

```diff
@@ -130,34 +130,38 @@ void StarField::Draw(const Point &pos, const Point &vel, double zoom, const Syst
 {
 	double density = system ? system->StarfieldDensity() : 1.;
 
-	double baseZoom = zoom;
-
 	// Check preferences for the parallax quality.
 	const auto parallaxSetting = Preferences::GetBackgroundParallax();
 	int layers = (parallaxSetting == Preferences::BackgroundParallax::FANCY) ? 3 : 1;
 	bool isParallax = (parallaxSetting == Preferences::BackgroundParallax::FANCY ||
 						parallaxSetting == Preferences::BackgroundParallax::FAST);
 
+	double baseZoom = isParallax ? zoom * 0.8 : zoom;
+
 	// Draw the starfield unless it is disabled in the preferences.
-	if(Preferences::Has("Draw starfield") && density > 0.)
+	for(int pass = 1; pass <= layers; pass++)
 	{
-		glUseProgram(shader.Object());
-		glBindVertexArray(vao);
+		Point backgroundPos = (isParallax ? (pos / 2) : pos) + Point(pass * Screen::Width(), pass * Screen::Height());
```

Comment:
```suggestion
		Point backgroundPos = (isParallax ? (pos / 2.) : pos) + Point(Screen::Width(), Screen::Height()) * pass;
```
It's possible this performs worse because of additional Point constructions, but I think it's easier to read.
Though, we could do something like:
```suggestion
		Point backgroundOffset(Screen::Width(), Screen::Height());
		backgroundOffset *= pass;
		Point backgroundPos = (isParallax ? (pos / 2.) : pos) + backgroundOffset;
```
Also, the division operator for Point takes a double.